cmake/external/onnx_minimal.cmake

```diff
@@ -46,10 +46,13 @@ endif()
 #     "${ONNX_SOURCE_ROOT}/onnx/test/*"
 #     "${ONNX_SOURCE_ROOT}/onnx/cpp2py_export.cc"
 # )
-# list(REMOVE_ITEM onnx_src ${onnx_exclude_src})  
-file(GLOB onnx_src CONFIGURE_DEPENDS
-"${ONNX_SOURCE_ROOT}/onnx/common/common.h"
-"${ONNX_SOURCE_ROOT}/onnx/defs/data_type_utils.*"
+# list(REMOVE_ITEM onnx_src ${onnx_exclude_src})
+set(onnx_src
+  "${ONNX_SOURCE_ROOT}/onnx/common/common.h"
+  "${ONNX_SOURCE_ROOT}/onnx/defs/data_type_utils.h"
+  "${ONNX_SOURCE_ROOT}/onnx/defs/data_type_utils.cc"
+  "${ONNX_SOURCE_ROOT}/onnx/defs/shape_inference.h"
+  "${ONNX_SOURCE_ROOT}/onnx/defs/shape_inference.cc"
 )
```

Comment:
Why do we need the shape inference code? 

I would have expected this causes a large hit to binary size but maybe most of it can be thrown away.

Reply:
graph.cc:MergeShapeInfo(): [https://github.com/microsoft/onnxruntime/blob/481b96d32aecc5fceabc9e9401604d2d31212ef7/onnxruntime/core/graph/graph.cc#L66](https://github.com/microsoft/onnxruntime/blob/481b96d32aecc5fceabc9e9401604d2d31212ef7/onnxruntime/core/graph/graph.cc#L66)
NodeArg::UpdateTypeAndShape()
api_impl.cc:ApiGraph::CopyValueInfo() [https://github.com/microsoft/onnxruntime/blob/481b96d32aecc5fceabc9e9401604d2d31212ef7/onnxruntime/core/optimizer/transpose_optimizer/api_impl.cc#L690](https://github.com/microsoft/onnxruntime/blob/481b96d32aecc5fceabc9e9401604d2d31212ef7/onnxruntime/core/optimizer/transpose_optimizer/api_impl.cc#L690)

the reason is ApiGraph::CopyValueInfo()

Reply:
removed the dependency on shape inference code